File: data/code_pairs/fortran/446.f90
module dpowint_mod
  implicit none
contains
  double precision function dpowint(xlow, xhigh, ylow, yhigh, xval, pow)
    implicit none
    double precision, intent(in) :: xlow, xhigh, ylow, yhigh, xval, pow
    double precision :: EPSLON
    EPSLON = 1.0D-20

    if (abs(xhigh-xlow) < EPSLON) then
      dpowint = (yhigh + ylow) / 2.0D0
    else
      dpowint = ylow + (yhigh - ylow) * &
                ((xval - xlow) / (xhigh - xlow))**pow
    endif
  end function dpowint
end module dpowint_mod

File: data/code_pairs/cpp/446.cpp
#include <cmath>

double dpowint(double xlow, double xhigh, double ylow, double yhigh, double xval, double pow) {
    const double EPSLON = 1.0e-20;

    if (std::abs(xhigh - xlow) < EPSLON) {
        return (yhigh + ylow) / 2.0;
    } else {
        return ylow + (yhigh - ylow) * std::pow((xval - xlow) / (xhigh - xlow), pow);
    }
}